hermes/bin/src/runtime_extensions/hermes/http_gateway/routing.rs

```diff
@@ -249,38 +275,112 @@ async fn handle_webasm_request(
     let (_parts, body) = req.into_parts();
     let body_bytes = body.collect().await?.to_bytes();
 
-    compose_http_event(
+    let request_params = HttpRequestParams {
         method,
         headers,
-        body_bytes,
+        body: body_bytes,
         path,
-        lambda_send,
-        &lambda_recv_answer,
-    )
+        sender: lambda_send,
+    };
+
+    compose_http_event(request_params, &lambda_recv_answer, module_id, &app_name)
 }
 
-/// Compose http event and send to global queue, await queue response and relay back to
-/// waiting receiver channel for HTTP response
-fn compose_http_event<B>(
+/// HTTP request parameters for WebAssembly event processing
+struct HttpRequestParams {
+    /// HTTP method (GET, POST, etc.)
     method: String,
+    /// HTTP headers as key-value pairs
     headers: HeadersKV,
+    /// Request body content
     body: Bytes,
+    /// Request path with query parameters
     path: String,
+    /// Channel sender for HTTP event responses
     sender: Sender<HTTPEventMsg>,
+}
+
+/// Composes and processes HTTP events for WebAssembly modules
+///
+/// This function handles the complete lifecycle of HTTP request processing:
+/// 1. Creates an HTTP event from request parameters
+/// 2. Determines target module(s) for routing
+/// 3. Sends event to the global queue
+/// 4. Waits for and processes the response
+fn compose_http_event<B>(
+    params: HttpRequestParams,
     receiver: &Receiver<HTTPEventMsg>,
+    module_id: Option<String>,
+    app_name: &ApplicationName,
 ) -> anyhow::Result<Response<B>>
 where
     B: Body + From<Vec<u8>> + From<String>,
 {
-    let on_http_event = HTTPEvent {
-        headers,
-        method,
-        path,
-        body,
-        sender,
+    let http_event = create_http_event(params);
+    let target_module = resolve_target_module(module_id, app_name)?;
+    let hermes_event = HermesEvent::new(http_event, TargetApp::All, target_module);
```

Comment:
Should this be `let hermes_event = HermesEvent::new(http_event, app_name, target_module);`